newdle/client/src/components/creation/EditPage.js

```diff
@@ -0,0 +1,51 @@
+import React, {useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Redirect, Route, Switch, useParams} from 'react-router-dom';
+import {Loader} from 'semantic-ui-react';
+import {abortCreation, fetchNewdle} from '../../actions';
+import {getCreatedNewdle, isLoggedIn} from '../../selectors';
+import {usePageTitle} from '../../util/hooks';
+import FinalStep from './FinalStep';
+import ParticipantsStep from './ParticipantsStep';
+import TimeslotsStep from './timeslots';
+
+export default function EditPage() {
+  const {code: newdleCode} = useParams();
+  const isUserLoggedIn = useSelector(isLoggedIn);
+  const newdle = useSelector(getCreatedNewdle);
+  const dispatch = useDispatch();
+  usePageTitle('Editing newdle');
+
+  useEffect(() => {
+    if (newdleCode) {
+      // TODO: We should re-use the existing newdle data instead. However, abortCreation will clear the state,
+      //  causing an edit -> summary -> edit to become empty if we remove the line below
+      //  Ideally, we should move abortCreation() - state flushing - to happen on mount instead of unmount.
+      dispatch(fetchNewdle(newdleCode, true));
+    }
```

Comment:
When reloading the page on an edit URL this loads the newdle twice since the SummaryHeader also triggers a fetch. I see two options here to avoid this:

- Do not use `SummaryHeader` for editing but a new `EditHeader`. This component could simply render the `SummaryHeader`; since it's different nesting I'd expect it to be a fresh render (ie the `SummaryHeader` will fetch the newdle again
- While it was me originally suggesting to always load fresh data, an alternative would be to just remove it here and assume people do not keep the summary page open for an extended time *and* make edits to their newdle from another tab/browser/device in the meantime.

Reply:
Exactly, this is something I would like to improve as the notes above. I think the problem is that we're consistently refreshing the global state, while the purpose of redux here should be the exact opposite.

IMO, we should refactor the way we handle state just like you suggested in the second point and ideally find a better place to load it rather than the header (because we have less control there). However, since I'm not sure about the impact of this change, I would suggest having it in a separate PR - and keeping the loads in this one.